velox/vector/tests/utils/VectorMaker.h

```diff
@@ -588,23 +588,83 @@ class VectorMaker {
         arrays.push_back(std::nullopt);
         continue;
       }
-
       std::vector<std::optional<T>> elements;
-      for (const auto& element : arrayObject) {
-        if (element.isNull()) {
-          // Null element.
-          elements.push_back(std::nullopt);
-        } else {
-          elements.push_back(detail::jsonValue<T>(element));
-        }
-      }
-
+      appendElementsFromJsonArray(arrayObject, elements);
       arrays.push_back(elements);
     }
-
     return arrayVectorNullable<T>(arrays, arrayType);
   }
 
+  template <typename T>
+  void appendElementsFromJsonArray(
```

Comment:
I suggest to move this new helper into the private section. It doesn't seem useful on its own to be used externally.